flink-state-backends/flink-statebackend-changelog/src/test/java/org/apache/flink/state/changelog/StateChangeLoggerTestBase.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.state.changelog;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.runtime.state.KeyGroupRange;
+import org.apache.flink.runtime.state.changelog.SequenceNumber;
+import org.apache.flink.runtime.state.changelog.StateChangelogWriter;
+import org.apache.flink.runtime.state.heap.InternalKeyContextImpl;
+import org.apache.flink.state.changelog.AbstractStateChangeLogger.StateChangeOperation;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static org.apache.flink.state.changelog.AbstractStateChangeLogger.StateChangeOperation.METADATA;
+import static org.junit.Assert.assertEquals;
+
+abstract class StateChangeLoggerTestBase<Namespace> {
+    /** A basic test for appending the metadata on first state access. */
+    @Test
+    public void testMetadataOperationLogged() throws IOException {
+        TestingStateChangelogWriter writer = new TestingStateChangelogWriter();
+        InternalKeyContextImpl<String> keyContext =
+                new InternalKeyContextImpl<>(KeyGroupRange.of(1, 1000), 1000);
+        StateChangeLogger<String, Namespace> logger = getLogger(writer, keyContext);
+
+        List<Tuple2<Integer, StateChangeOperation>> expectedAppends = new ArrayList<>();
+        expectedAppends.add(
+                Tuple2.of(-1 /* see AbstractStateChangeLogger#COMMON_KEY_GROUP */, METADATA));
```

Comment:
How about make `AbstractStateChangeLogger#COMMON_KEY_GROUP` package public so that we coule refer to that parameter directly instead of a magic number here.

Reply:
Good idea.

Reply:
It seems you did not change the magic number to `COMMON_KEY_GROUP` here in test code.

Reply:
Yes, sorry (I changed it but then accidentially rolled back together with type changes for other comment).